repository/dimensions_repository.go

```diff
@@ -0,0 +1,132 @@
+package repository
```

Comment:
Fair point, hadn't appreciated how many places were using them. Additional benefit of models package that I also hadn't appreciated last week - allows external libraries to import the models package without requiring a bunch of new dependencies. For the repository ones, I think you're right, they're shared and should stay separate. 

The message ones I'm less certain of, to me they fit quite cleanly under the message package - gives useful context as well when it's called from handlers eg `message.DimensionsExtractedEvent`. But not precious about it, if you don't think it adds useful context dont worry about changing it :)